### PR TITLE
Added test case to show the bug described in #1853

### PR DESCRIPTION
**Pull request by duelinggalois**
Test case reproduces the error described in #1853. I found a quick and dirty hack to fix it locally, but it looks like my hack does not work in the test suite so I did not add it.

### CODE REVIEW

File: slither/solc_parsing/slither_compilation_unit_solc.py [hunk 1]
```diff
@@ -742,6 +742,14 @@ def _convert_to_slithir(self) -> None:
                     self._underlying_contract_to_parser[contract].log_incorrect_parsing(
                         f"Impossible to generate IR for {contract.name}.{func.name} ({func.source_mapping}):\n {e}"
                     )
+                except AssertionError as e:
+                    # Provide contex to user about code that failed beyond AssertionError and stack trace.
+                    func_expressions = "\n".join([f'\t{ex}' for ex in func.expressions])
+                    logger.error(
+                        f"Failed to parse {contract.name}.{func.name} ({func.source_mapping}):\n "
+                        f"{func_expressions}")
+                    raise e
```
Comment (0xalpharush) [inline slither/solc_parsing/slither_compilation_unit_solc.py:751]: This looks really helpful and is a great idea! Would you mind making a PR with just these changes?

File: slither/solc_parsing/slither_compilation_unit_solc.py [hunk 2]
```diff
@@ -742,6 +742,14 @@ def _convert_to_slithir(self) -> None:
                     self._underlying_contract_to_parser[contract].log_incorrect_parsing(
                         f"Impossible to generate IR for {contract.name}.{func.name} ({func.source_mapping}):\n {e}"
                     )
+                except AssertionError as e:
+                    # Provide contex to user about code that failed beyond AssertionError and stack trace.
+                    func_expressions = "\n".join([f'\t{ex}' for ex in func.expressions])
+                    logger.error(
+                        f"Failed to parse {contract.name}.{func.name} ({func.source_mapping}):\n "
```
Comment (0xalpharush) [inline slither/solc_parsing/slither_compilation_unit_solc.py:749]:
```suggestion
                        f"\nFailed to parse {contract.name}.{func.name} ({func.source_mapping}):\n "
```

File: slither/solc_parsing/slither_compilation_unit_solc.py [hunk 3]
```diff
@@ -742,6 +742,14 @@ def _convert_to_slithir(self) -> None:
                     self._underlying_contract_to_parser[contract].log_incorrect_parsing(
                         f"Impossible to generate IR for {contract.name}.{func.name} ({func.source_mapping}):\n {e}"
                     )
+                except AssertionError as e:
+                    # Provide contex to user about code that failed beyond AssertionError and stack trace.
```
Comment (0xalpharush) [inline slither/solc_parsing/slither_compilation_unit_solc.py:746]:
```suggestion
```

File: slither/solc_parsing/slither_compilation_unit_solc.py (hunk 1 above)
Comment (duelinggalois) [reply slither/solc_parsing/slither_compilation_unit_solc.py:751]: done #1873

**Pull request closed without merging**